js/index.js

```diff
@@ -1,7 +1,60 @@
+  } else if (hacker1.length<hacker2.length) {
+  console.log(`It seems that the navigator has the longest name, it has ${hacker2.length} characters.`) 
+  } else {
+  console.log(`Wow, you both have equally long names, ${hacker1.length} characters!`) 
+  }
 
+//iteration 3
+//3.1
+let str=' '
+for   (let i=0; i<hacker1.length; i++){
+  str += hacker1.charAt(i).toUpperCase() + ' ';
+}
+console.log(str)
 
-// Iteration 3: Loops
\ No newline at end of file
+//const separated= hacker1.match(/.{1}/g) --›encontrado online pero no entiendo que quiere decir la syntaxis
+
+//const separated= hacker1.match(/.{1}/g).join(" ").toUpperCase()
+//console.log(separated) --> RESULTADO OK!
+
+
+//3.2
+let str2=' '
+for   (let i=hacker2.length; i>=0; i--){
+  str2 += hacker2.charAt(i) + ' ';
+}
+console.log(str2)
+
+//3.3
+if (hacker1<hacker2){
+    console.log("The driver's name goes first.")
+    } else if (hacker1>hacker2){console.log("Yo, the navigator goes first definitely.")
+    } else {console.log("What?! You both have the same name?")
+    }
+
+//Bonus 1 
+const lorem= "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Proin egestas tincidunt venenatis. Pellentesque ut ipsum vel orci maximus lacinia a id est. Vivamus consequat sit amet urna vel molestie. Vivamus dictum bibendum purus nec dapibus. Phasellus congue fermentum velit a finibus. Quisque ac ex convallis, fermentum orci vitae, consequat mauris. Mauris ut feugiat leo. Integer arcu lectus, placerat vitae lorem ut, congue auctor ligula. Morbi aliquam arcu sed ligula pretium, vel elementum tellus maximus. Ut non efficitur erat, eget pellentesque massa. Duis sodales id orci dignissim ornare. Aenean auctor libero ut semper ultrices. Curabitur imperdiet eleifend nunc sit amet tempor. Suspendisse ac feugiat nulla. Fusce tempus lorem lectus, sit amet elementum nisl luctus et. Quisque iaculis tempor erat sed laoreet. Aenean efficitur finibus orci, sed fermentum nisi ullamcorper ac. Proin tristique laoreet volutpat. Mauris vitae nisl vel ante faucibus efficitur. Pellentesque imperdiet felis tortor, id porttitor odio lobortis eget. Etiam laoreet eu felis ut egestas. Sed non sodales odio. Morbi nisi leo, suscipit quis ultricies ac, tincidunt ut mauris. Etiam blandit bibendum velit vel hendrerit. Morbi molestie maximus nisl nec tempus. Nunc porttitor pulvinar dui eget pretium. Quisque congue imperdiet est, eu blandit arcu tristique ac. Maecenas tortor dui, dignissim sed hendrerit vitae, aliquet porttitor felis. Aenean sit amet facilisis turpis. Ut vel sagittis nisi. Quisque hendrerit eu erat ut porttitor. Nulla at tellus fringilla, dignissim justo sit amet, tristique eros. Fusce facilisis magna massa, ac molestie dolor iaculis placerat."
+console.log(lorem.split(" ").length)
+console.log(lorem.match(/et/g).length)
+
+
+//Bonus 2
+//Por favor podeis explicar este ejercicio en clase? Hemos encontrado esta solucion en Internet, pero no funciona, y hemos intentado mil soluciones mas y nada nos funciona. 
```

Comment:
Para aplicar esta lógica podemos tener en cuenta DOS casos.

Uno en el que no hay signos de puntuación, exclamación, ni caractéres raros que comparar... en ese caso, es coger nuestro string, unirla y ver si a la inveresa es igual.

Dos, el caso complejo que es el que habéis tratado de hacer, que es cuando tenemos signos de puntuación o caracteres extraños:

1. Creo una variable que sea igual a lo que escribamos en el prompt, quitándole todos los caracteres raros, para hacer esto puedo usar la función replace, pasando una REGEX, que es este string ` /[\s,.!¡¿?:;']/g, ''` que lo hace es buscar caracteres como `!,.,¿,?,:,...` y los reemplaza por `string` vacío `''` así conseguimos que por ejemplo `!ana!, ana?`, se nos quede en la palabra o frase que queremos comprobar.
También va concatenada otra regex que lo que hace es quitarle los acentos a las palabras, para que también no intercedan a la hora de comprobarlo. Después de limpiar la frase de este tipo de caracteres lo pasamos todo a minúsculas para que también no intercedan las mayúsculas a la hora de comparar. Esto además quitarría también los espacios en blanco. Quedaría así:

Por ejemplo la frase: `A mamá Roma, le aviva el amor a papá y a papá Roma le aviva, el amor a mamá.`

```
const isPalindrome = prompt("Let's write a palindrome, we will check it!").replace(/[\s,.!¡¿?:;']/g, '').normalize("NFD").replace(/[\u0300-\u036f]/g, "").toLowerCase()
```

Esto daría: `amamaromaleavivaelamorapapayapaparomaleavivaelamoramama`

2. Ahora solo quedaría comprobar que esta frase y su invertida son iguales:

```
  if (isPalindrome == isPalindrome.split('').reverse().join('')){
   console.log('The sentence you wrote, is a palindrome!' + '\n')
 } else {
   console.log('Sorry its not a palindrome!' + '\n')
 }
```
Esto mismo por ejemplo podríais hacerlo sin REGEX de manera un poco más enrevesada, pero alomejor más fácil de entender:

```
 var isPalindrome = prompt("Let's write a palindrome, we will check it!");
 var arrayPalindrome = [];
 
 for (var i = 0; i <= isPalindrome.length -1; i++ ){
   switch(isPalindrome[i]){
   case" ":case".":case",":case"!":case"¡":case"*":case"'":case'"':case"@":
   case"#":case"%":case"&":case"(":case")":case"¿":case"?":case":":case"$":
   case"ª":case"º":case"-":case"_":case";":case"·":case"/":case"=":
    arrayPalindrome.push('');
    break;
    default:
    arrayPalindrome.push(isPalindrome[i].toLowerCase());
     }
 }
```

Creo un swicth por ejemplo que cada vez que se encuentre con un carácter extraño lo sustituya por un string vacío, pero es un switch largo y complejo que no tiene mucho sentido... para el mismo resultado podemos utilizar la regex, que lo hace por nosotros. Por eso son útiles!